### ISSUE TITLE
cargo doesn't resolve dependency with relative path correctly

### ISSUE THREAD

**Issue by brenzi** [rust-lang/cargo#6657]
**Problem**
I have relative paths to a git submodule in my Cargo.toml dependencies. In one odd case the path is not resolved correctly and cargo complains 

```
error: failed to load source for a dependency on `serde`

Caused by:
  Unable to update /home/brenzi/encointer/third_party/serde-rs/serde/serde

Caused by:
  failed to read `/home/brenzi/encointer/third_party/serde-rs/serde/serde/Cargo.toml`

Caused by:
  No such file or directory (os error 2)

```

So `../../` seems to jump one hierarchy too much or the path is overridden somewhere.

**Steps**

1. `git clone https://github.com/brenzi/substraTEE.git`
2. `cd substraTEE/wasm-sgx`
3. `git checkout 3f3339f7cda15a13e04229cd56fbe0ee1344e464`
4. `make`

or 
4. `cd enclave`
5. `cargo build`

basically I just moved samplecode from
`substraTEE/rust-sgx-sdk/samplecode/wasmi` to `substraTEE/wasm-sgx`

**Possible Solution(s)**
I already saw that xargo.toml files paths need to be fixed, however, fixing them didn't change the behaviour.

I wonder if symlinks in serde/serde folder cause trouble

**Notes**

Output of `cargo version`:
cargo 1.34.0-nightly (865cb7010 2019-02-10)
ubuntu 16.04

**Comment by ehuss** [comment]
The paths in the sgxwasm crate have too many `..` here: https://github.com/brenzi/substraTEE/blob/3f3339f7cda15a13e04229cd56fbe0ee1344e464/wasm-sgx/enclave/sgxwasm/Cargo.toml#L9

It would be nice if Cargo reported where the offending dependency is specified.  That shouldn't be too hard to add.

`sgx_types` is also missing from your repo.

**Comment by brenzi** [comment]
sorry, the problem was another Cargo.toml file in a subdirectory being referenced....
